framework/include/base/MooseFunctor.h

```diff
@@ -917,15 +920,21 @@ class FunctorEnvelope final : public FunctorBase<T>, public FunctorEnvelopeBase
   /**
    * @param wrapped The functor to wrap. We will *not* not own the wrapped object
    */
-  FunctorEnvelope(const FunctorBase<T> & wrapped) : FunctorEnvelopeBase(), _wrapped(&wrapped) {}
+  FunctorEnvelope(const FunctorBase<T> & wrapped)
+    : FunctorBase<T>("wraps_" + wrapped.functorName()), FunctorEnvelopeBase(), _wrapped(&wrapped)
+  {
+  }
 
   /**
    * @param wrapped A unique pointer around the functor to wrap. We *will* own the wrapped object,
    * e.g. if we are ever destructed or we are reassigned to wrap another functor, then this functor
    * will be destructed
    */
   FunctorEnvelope(std::unique_ptr<FunctorBase<T>> && wrapped)
-    : FunctorEnvelopeBase(), _owned(std::move(wrapped)), _wrapped(_owned.get())
+    : FunctorBase<T>("wraps_" + wrapped->functorName()),
```

Comment:
I'm guessing that a consequence of this is that the name of this wrapper functor is different than its stored name on the `subproblem`?

Reply:
No, I don't. I like the idea of globally unique names as much as possible. Because for PINSFV for example we essentially have three porosities: `wraps_porosity`, `porosity`, and `smoothed_eps` (now that I see the last one, maybe we should rename that functor). And I don't think we want to be renaming `wraps_porosity` when we swap out `porosity` for `smoothed_eps`. I think the globally unique names for these three different functors looks really good here.

My thought would be that in the places we do `functors.emplace` and `functors.find` in `Subproblem.h`, we query with a `wrapped_name` where we take the incoming `name` and add a `wraps_` prefix. When we create a new letter functor we use `name`, but when we search and/or create a new envelope functor, we use `wrapped_name`. Then the `_functors` map keys will match the `functorName()` of the map values.

What do you think? The user gets to work with non-prefixed names and we get our globally unique functor naming schemes.

Reply:
If you do this modification, I think the only other code you should modify is this code
```c++
void
SubProblem::initialSetup()
{
  for (const auto & functors : _functors)
    for (const auto & pr : functors)
      if (pr.second->wrapsNull())
        mooseError("No functor ever provided with name '",
                   pr.first,
                   "', which was requested by '",
                   MooseUtils::join(libmesh_map_find(_functor_to_requestors, pr.first), ","),
                   "'.");
}
```
I would make sure the remove the `wraps_` prefix before erroring

Reply:
Pfeww that was actually hard to get it right. Hopefully I did though. Let's see what test suite says